corehq/form_processor/document_stores.py

```diff
@@ -46,3 +51,46 @@ def iter_document_ids(self, last_id=None):
     def iter_documents(self, ids):
         for wrapped_case in self.case_accessors.iter_cases(ids):
             yield wrapped_case.to_json()
+
+
+class ReadonlyLedgerDocumentStore(ReadOnlyDocumentStore):
+
+    def __init__(self, domain):
+        assert should_use_sql_backend(domain), "Only SQL backend supported"
+        self.domain = domain
+        self.ledger_accessors = LedgerAccessorSQL
+        self.case_accessors = CaseAccessorSQL
+
+    def get_document(self, doc_id):
+        from corehq.form_processor.parsers.ledgers.helpers import UniqueLedgerReference
+        try:
+            ref = UniqueLedgerReference.from_id(doc_id)
+            return self.ledger_accessors.get_ledger_value(**ref._asdict()).to_json()
+        except LedgerValueNotFound as e:
+            raise DocumentNotFoundError(e)
+
+    @property
+    @memoized
```

Comment:
since the document store gets reinitialized with each change it might make sense to make this a quickcached function instead of a memoized property in case we want to optimize for processing several changes in a domain in rapid succession.